src/id.rs

```diff
@@ -146,98 +152,121 @@ impl Display for PublicId {
-        let (age, public_sign_key, public_encrypt_key): (u8, PublicSignKey, PublicEncryptKey) =
+        let (age, public_signing_key, public_encryption_key) =
             Deserialize::deserialize(deserialiser)?;
-        Ok(PublicId::new(age, public_encrypt_key, public_sign_key))
+        Ok(PublicId::new(
+            age,
+            public_signing_key,
+            public_encryption_key,
+        ))
     }
 }
 
 impl parsec::PublicId for PublicId {
-    type Signature = Signature;
+    type Signature = signing::Signature;
+
     fn verify_signature(&self, signature: &Self::Signature, data: &[u8]) -> bool {
-        self.signing_public_key().verify_detached(signature, data)
+        self.verify(data, signature)
     }
 }
 
 impl PublicId {
-    /// Return age.
+    /// Returns age.
     pub fn age(&self) -> u8 {
         self.age
     }
 
-    /// Return initial/relocated name.
+    /// Returns initial/relocated name.
     pub fn name(&self) -> &XorName {
         &self.name
     }
 
-    /// Return public encrypting key.
-    pub fn encrypting_public_key(&self) -> &PublicEncryptKey {
-        &self.public_encrypt_key
+    /// Verifies this id signed a message
+    pub fn verify(&self, message: &[u8], sig: &signing::Signature) -> bool {
+        self.public_signing_key.verify(message, sig).is_ok()
     }
 
-    /// Return public signing key.
-    pub fn signing_public_key(&self) -> &PublicSignKey {
-        &self.public_sign_key
+    /// Returns public signing key.
+    pub fn public_signing_key(&self) -> &signing::PublicKey {
+        &self.public_signing_key
+    }
+
+    /// Returns public encryption key.
+    pub fn public_encryption_key(&self) -> &encryption::PublicKey {
+        &self.public_encryption_key
     }
 
     fn new(
         age: u8,
-        public_encrypt_key: PublicEncryptKey,
-        public_sign_key: PublicSignKey,
+        public_signing_key: signing::PublicKey,
+        public_encryption_key: encryption::PublicKey,
     ) -> PublicId {
         PublicId {
             age,
-            name: Self::name_from_key(&public_sign_key),
-            public_sign_key,
-            public_encrypt_key,
+            name: name_from_key(&public_signing_key),
+            public_signing_key,
+            public_encryption_key,
         }
     }
+}
 
-    fn name_from_key(public_sign_key: &PublicSignKey) -> XorName {
-        XorName(safe_crypto::hash(&public_sign_key.into_bytes()))
-    }
+fn name_from_key(public_key: &signing::PublicKey) -> XorName {
+    XorName(public_key.to_bytes())
+}
+
+#[cfg(not(any(test, feature = "mock_base")))]
+fn gen_secret_signing_key() -> signing::SecretKey {
+    use rand::OsRng;
+    let rng = OsRng::new().expect("Cannot generate random data, unsafe to continue");
+    signing::SecretKey::generate(&mut RngCompat(rng))
+}
+
+#[cfg(any(test, feature = "mock_base"))]
+fn gen_secret_signing_key() -> signing::SecretKey {
+    use maidsafe_utilities::SeededRng;
+    let rng = SeededRng::thread_rng();
```

Comment:
Does that ensure reproducible test seeds?

Reply:
It does. `SeededRng` uses a global shared seed so as long as you create one `SeededRng` instance using `from_seed` somewhere at the beginning of your tests, all the other instances will be created from that seed.